### ISSUE TITLE
"Duplicate" fails when single sample selected

### ISSUE THREAD

**Issue by SteveDaulton** [audacity/audacity#6009]
### Bug description

"Ctrl + D" (Duplicate) fails when a single sample is selected.

### Steps to reproduce

1. Generate a tone
2. Zoom in to see individual samples
3. Select one  sample
4. Duplicate

### Expected behavior

The selected sample to be duplicated in a new track.

### Actual behavior

A new track is created, and a 1 sample wide region is selected in the new track, but the track is empty (no samples)

### Audacity Version

current master / alpha / nightly build

### Operating system

Linux

**Comment by SteveDaulton** [comment]
It also appears to be impossible to copy / paste a single sample.

**Comment by petersampsonaudacity** [comment]
Confirmed on W10 - but this is surely an extreme edge-case ...

**Comment by SteveDaulton** [comment]
>this is surely an extreme edge-case

Sure, but "off by one" issues can cause much bigger problems, and at present no-one knows the cause of this bug or whether it can cause more serious issues.

**Comment by LWinterberg** [comment]
indeed, duplicating 5 samples shows a new clip with 4 samples, so the off-by-one-ness continues for all duplicates. However, the trims are non-destructive so at least there's no data loss

**Comment by petersampsonaudacity** [comment]
The problem would appear to be with Selection rather than the duplication or copying.

Look very closely at this example and observe this short selection:
![image](https://github.com/audacity/audacity/assets/50205984/4f9a72f1-92c1-465f-b687-44754e8f5412)

1. even with Snap To off you can't select in between samples
2. casual observation makes it look as though the fourth sample is included - but close observation shows that this is NOT the case, the selection stops just shy of it
3. this is confirmed by examining the Selection toolbar which is set to samples and start/length

-------------------------------------------------------------------------------------

Curiously if you generate a 30 second chirp you get1,323,000 samples  - and if you
a) duplicate it, or
b) Crtrl+A and copy it to a new track
in both cases you get the full 1,323,000 samples as a result.

**Comment by petersampsonaudacity** [comment]
This selection behavior was introduced back in Audacity 2.1.2 - this was when the display of the samples was changed to a bigger more visible dot.

In 2.1.1 and earlier it was possible (only possible in fact) to select between the samples - see this example from 2.1.1
![image](https://github.com/audacity/audacity/assets/50205984/74b3347f-6d19-428c-b36d-bef80693e633)
where 
a) the selection extends to half-way between samples
b) 4 samples can be counted within the bounds of the selection display

So effectively this is a regression on 2.1.1 and earlier

**Comment by SteveDaulton** [comment]
It is also a regression on Audacity 2.4.2 and Audacity 3.3.2. Both of these versions have the new way to represent sample selections, but do not have this bug.
